src/lib/y2partitioner/widgets/fstab_options.rb

```diff
@@ -337,6 +348,10 @@ def label
 
       def store
         mount_point.mount_by = selected_mount_by
+        filesystem.copy_mount_by_to_subvolumes if btrfs?
```

Comment:
I'm not sure if having to call `copy_mount_by_to_subvolumes` every time we change the mount_by of the Btrfs is robust enough. I would sleep better if that would be a responsibility of `MountPoint#mount_by=`.

99.99% of the times you want the Btrfs to be on sync with the subvolumes, so I think it's better if the models already guarantee that implicitly. Maybe is more cumbersome in the (quite weird) case in which you really want to enforce the Btrfs and its subvolumes to be out of sync (if that's technically safe) but... Do we want to make this more error prone for the regular case just to make the exception easier?

Reply:
On the other hand, the idea of having a separate `Action` (even a separate `Controller` in this case of a multi-step action) is to have clean widgets with as little logic as possible. So the business logic is not scattered through the UI code.

So I would suggest something like:
```ruby
  def store
    controller.mount_by = selected_mount_by
  end
```

And taking the decisions in the controller:

```ruby
def mount_by=(value)
  mount_point.mount_by = value
  filesystem.copy_mount_by_to_subvolumes if btrfs?
end
```

So the widget stays as a widget, and not as an intelligent thingie that needs to know whether it's in a btrfs context or not and act in consequence.

Reply:
So you think that hiding that snippet in the controller is a better approach?

I disagree. Nobody will ever look there. Everybody is just looking at the place where the user changes the value, and that's in the widget.

Right now, we have the relevant code scattered among a half-dozen places; this is what took me so long to figure out where to put what. Adding yet another place that will surprise anybody doing maintenance on the code is clearly not the way to go IMHO.

Add to that the fact that this particular controller is way too large and way too complex already - it is the one controller that serves 3 or 4 different dialogs already. We already talked about the need to split it up, so the places where the relevant code is scattered will multiply even more.